Deprovisioning (#14)
kyma-project/btp-manager: docs/deprovisioning.md

## Deprovisioning

To start the deprovisioning process, use the following command:

```

> kubectl delete btpoperator yourbtpoperator

Reviewer: ```suggestion
kubectl delete btpoperator your-btpoperator
```

```

The command triggers deletion of all service bindings, service instances and BtpOperator resources on your cluster.

> After the deletion of deployment and webhooks which `yourbtpoperator` manages, the deprovisioning flow tries to perform deletion in hard delete mode. When it finds all service bindings and their secrets, and service instances across all namespaces, it tries to delete Kubernetes.

Reviewer: ```suggestion
After the deletion of deployment and webhooks which `your-btpoperator` manages, the deprovisioning flow tries to perform deletion in hard delete mode. When it finds all service bindings and their secrets, and service instances across all namespaces, it tries to delete Kubernetes.
```

The time limit for this operation is 20 minutes.

> After this time, or in case of error in hard deletion, the system will go into soft delete mode which mean deletion of finalizers from service instances and bindings.

Reviewer: ```suggestion
After this time, or in case of an error in hard deletion, the system goes into soft delete mode, which runs deletion of finalizers from service instances and bindings.
```

> Regardless of mode, in the next step, the all BtpOperator resources marked with the "managed-by:btp-operator" label are deleted.

Reviewer: ```suggestion
Regardless of mode, in the next step, all the BtpOperator resources marked with the "managed-by:btp-operator" label are deleted.
```

> Regardless of mode, at later step, the all btp operator resources marked with label "managed-by:btp-operator" are deleted.

Reviewer: ```suggestion
Regardless of mode, in the next step, the all BtpOperator resources marked with the "managed-by:btp-operator" label are deleted.
```

> If proccess pass with success then finally finalizer on yourbtpoperator itself is removed and the resource is deleted.

Reviewer: ```suggestion
If the process runs successfully, the finalizer on `yourbtpoperator` itself is removed and the resource is deleted.
```

> If the process runs successfully, the finalizer on `yourbtpoperator` itself is removed and the resource is deleted.
> If an error occurs during deprovisioning, `yourbtpoperator` is set to `Error`.

Reviewer: ```suggestion
If the process runs successfully, the finalizer on `your-btpoperator` itself is removed and the resource is deleted.
If an error occurs during deprovisioning, `your-btpoperator` is set to `Error`.
```

> If any error will happen during deprovisioning, then yourbtpoperator will be set to Error state.

Reviewer: ```suggestion
If an error occurs during deprovisioning, `yourbtpoperator` is set to `Error`.
```

> ![KEB diagram](./assets/keb-architecture.svg)

Reviewer: ```suggestion
![Deprovisioning diagram](./assets/deprovisioning.svg)
```

> ![KEB diagram](./assets/keb-architecture.svg)

Reviewer: In the diagram, is it `Reconcile` or `Reconciler`?
